secure-api-gateway-ob-uk-rs-validation/secure-api-gateway-ob-uk-rs-validation-obie/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/validation/obie/v4/payment/BasePaymentRequestValidator.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright © 2020-2024 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.ob.uk.rs.validation.obie.v4.payment;
+
+import java.util.Objects;
+import java.util.function.Supplier;
+
+import com.forgerock.sapi.gateway.ob.uk.common.error.OBRIErrorType;
+import com.forgerock.sapi.gateway.ob.uk.rs.validation.ValidationResult;
+import com.forgerock.sapi.gateway.ob.uk.rs.validation.obie.BaseOBValidator;
+import com.forgerock.sapi.gateway.ob.uk.rs.validation.obie.v4.payment.BasePaymentRequestValidator.PaymentRequestValidationContext;
+
+import uk.org.openbanking.datamodel.v3.error.OBError1;
```

Comment:
I think this is a reasonable compromise for v4, to continue to use OBError1 from v3 pkg because all exception handling code currently uses this and the schema is the same except changing the code field to be a 4 letter code.

In future we may need to revisit this, in which case it will make sense to introduce our own Error class which can then be converted into v3 and v4 response types in the exception handler. This will be a fairly large piece of work, which is why I think it makes sense to avoid it for now (as there is no functional benefit at the moment).

One thing which needs to be considered is how the OBError1.code value is populated in the response, as in v4 the values are now 4 letter codes e.g. v3 code: `UK.OBIE.Field.Expected` becomes: `U001` - this is a simple mapping according to values here: https://github.com/OpenBankingUK/External_Internal_CodeSets/blob/main/OB_Internal_CodeSet.csv

For ease of migration this translation can be added in the GlobalExceptionHandler - this handler is cross cutting but I expect it has access to the request obj and can therefore determine the API version being used, then it can translate the error code as needed. By convention, the validators can create OBError1 objects with v3 codes, translation is only required for v4 responses.

@mariantiris can you update the v4 functional tests to assert these new error codes please?

FYI - @BohoCode - let me know if you want to discuss this in more detail.